code_generator/End2End/GeneratedDependencies/Generated/improbable/testschema/ExhaustiveBlittableSingularTranslation.cs

```diff
@@ -168,7 +168,136 @@ public void OnComponentUpdate(ComponentUpdateOp<global::Improbable.TestSchema.Ex
                 }
 
                 componentData.DirtyBit = false;
-                view.UpdateComponent(entity, componentData, UpdatesPool);
+
+                view.SetComponentData(entity, componentData);
+
+                var componentFieldsUpdated = false;
+                if (update.field1.HasValue)
```

Comment:
Please combine this with the `if`s above

Reply:
Combined the two code blocks. We now preemptively make an update struct without knowing whether there will be field updates but that's probably fine.